lib/xdr/bool.ex

```diff
@@ -1,78 +1,78 @@
 
+  @typedoc """
+  `XDR.Bool` struct type specification.
+  """
   @type t :: %XDR.Bool{declarations: keyword(), identifier: boolean()}
 
-  @spec new(atom()) :: t()
-  def new(identifier), do: %XDR.Bool{declarations: @boolean, identifier: identifier}
+  @doc """
+  Create a new `XDR.Bool` structure from the `identifier` passed either `false` or `true`.
+  """
+  @spec new(identifier :: atom()) :: t()
+  def new(identifier), do: %XDR.Bool{identifier: identifier}
 
   @impl XDR.Declaration
   @doc """
-  This function is in charge of encode the boolean data to a binary representation
-
-  returns an ok tuple with the boolean encoded into an XDR value
+  Encode a `XDR.Bool` structure into a XDR format.
   """
-  @spec encode_xdr(t()) :: {:ok, binary} | {:error, :not_boolean}
+  @spec encode_xdr(boolean :: t()) :: {:ok, binary} | {:error, :not_boolean}
   def encode_xdr(%XDR.Bool{identifier: identifier}) when not is_boolean(identifier),
     do: {:error, :not_boolean}
 
   def encode_xdr(%XDR.Bool{} = boolean), do: Enum.encode_xdr(boolean)
 
   @impl XDR.Declaration
   @doc """
-  This function is in charge of encode the boolean data to a binary representation
-
-  returns the boolean encoded into an XDR value
+  Encode a `XDR.Bool` structure into a XDR format.
+  If the structure received is not `XDR.Bool` or the identifier is not boolean, an exception is raised.
   """
-  @spec encode_xdr!(t()) :: binary()
+  @spec encode_xdr!(boolean :: t()) :: binary()
   def encode_xdr!(boolean) do
     case encode_xdr(boolean) do
       {:ok, binary} -> binary
-      {:error, reason} -> raise(Bool, reason)
+      {:error, reason} -> raise(BoolError, reason)
     end
   end
 
   @impl XDR.Declaration
   @doc """
-  This function is in charge of decode XDR data which represents a boolean value
-
-  returns an ok tuple with the boolean decoded from an XDR value
+  Decode the Boolean in XDR format to a `XDR.Bool` structure.
   """
   @spec decode_xdr(bytes :: binary, struct :: t | any) ::
           {:ok, {t, binary}} | {:error, :invalid_value}
-  def decode_xdr(bytes, struct \\ %XDR.Bool{declarations: @boolean})
+  def decode_xdr(bytes, struct \\ %XDR.Bool{})
   def decode_xdr(bytes, _struct) when not is_binary(bytes), do: {:error, :invalid_value}
 
   def decode_xdr(bytes, struct) do
     {enum, rest} = Enum.decode_xdr!(bytes, struct)
-
     decoded_bool = new(enum.identifier)
-
     {:ok, {decoded_bool, rest}}
   end
 
   @impl XDR.Declaration
   @doc """
-  This function is in charge of decode XDR data which represents a boolean value
-
-  returns the boolean decoded from an XDR value
+  Decode the Boolean in XDR format to a `XDR.Bool` structure.
+  If the binary is not a valid boolean, an exception is raised.
   """
-  @spec decode_xdr!(bytes :: binary, struct :: t | any) :: {t(), binary}
-  def decode_xdr!(bytes, struct \\ %XDR.Bool{declarations: @boolean})
+  @spec decode_xdr!(bytes :: binary, struct :: t() | any) :: {t(), binary}
```

Comment:
@marioyustikommit what is the meaning of t() as a typespec

Reply:
Sorry, I was investigating and I realized that It is a bad practice that I adopted. I fixed it.